### ISSUE TITLE
Add field set support to GAU Allocations

### ISSUE THREAD

**Issue by njjc** [SalesforceFoundation/Cumulus#1655]
https://powerofus.force.com/0D58000002CGoZp

**Comment by NorthPeakMelissa** [comment]
Just ran into this issue today. It would be a lovely feature.

Thanks!

**Comment by brianpickett** [comment]
Is this in progress? We may be able to pick it up...

**Comment by njjc** [comment]
It's not in progress, but I'm happy to collaborate if you're interested in
contributing!

**Comment by brianpickett** [comment]
Ok, thanks. We're poking around in the code over the next two weeks and will share some thoughts/examples afterwards as a basis for a discussion.

**Comment by njjc** [comment]
Sounds great. You could also participate in the NPSP sprint, in person or
remotely, which starts [three weeks from today](
http://www.salesforcefoundation.org/sprinting-for-the-nonprofit-starter-pack/),
and do the work then!

**Comment by judisohn** [comment]
Thank you, Brian! Looking forward to seeing what you come up with.

Sent from my iPhone

On Oct 1, 2015, at 5:30 PM, Brian Pickett <notifications@github.com> wrote:

Ok, thanks. We're poking around in the code over the next two weeks and
will share some thoughts/examples afterwards as a basis for a discussion.

—
Reply to this email directly or view it on GitHub
<https://github.com/SalesforceFoundation/Cumulus/issues/1655#issuecomment-144855852>
.

**Comment by kbromer** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x800000001fwNAAQ" target="_blank">W-008314</a>

**Comment by sd5000** [comment]
Yes, also just ran into this, mentioned to @kbromer. To add a few more fields to the Manage Allocations layout we had to duplicate the page, controller, and test code (which obviously isn't great practice) -- along with a couple associated UTIL classes -- so we could add a few fields to the SOQL query and table layout. Would be great to be able to use a field set instead.

**Comment by tet3** [comment]
I took a look at this since it came back up on the Hub, and I did the field set support for the Payments Wizard a while back. This is trickier, because there's JS attached to events on the Amount and Percent fields. The simplest way to implement this, I think, would be to leave the 3 fields there (GAU, %age, Amount) hardcoded into the VF and then have the fieldset be just for additional fields that would display to the right of Percent. Does that seem like it would meet users' needs and NPSP standards? 

Possible alternatives:  

- include these three fields in the field set but document that all three are required, and error on the page if one of them is missing. (I'm pretty sure, but not certain, that the `isRequired` property  in `displayedFields` in the object metadata doesn't actually prevent subscribers from removing the field from the fieldset. I think it just provides an attribute that can be queried for and used to mark a field as required in the UI.) 
- Similar to the above, but only require one of Amount or Percentage to be present. 

Either of these would give subscribers maximum flexibility in ordering the fields.  LMK if I should take this back to the [Idea](https://powerofus.force.com/s/idea/a1W80000006huGDEAY/allow-for-addition-of-fields-to-the-gau-allocation-vf-page) to get some Hubbers' input. 

FYI @coriobriensfdo @mpusto (sorry - don't see a way to tag Alison Meyer here!)

**Comment by sd5000** [comment]
@tet3 sounds good. I prefer the latter options that give increased flexibility for field order, including the option to remove Amount or Pct if appropriate. In our experience the teams who are doing allocation data entry have been understandably particular about optimizing their workflow, given the scale of records they have to maintain.

**Comment by coriobriensfdo** [comment]
@tet3 it certainly wouldn't hurt posting this in the Idea and getting additional feedback :)

Any chance you can make it to one of our upcoming Open Source Community Sprint's (Detroit in July or Philly in Oct)?

**Comment by stephaniepck** [comment]
Hi everyone.  I know this has been an older post and ongoing but wanted to see if anyone figured anything out here.  We're trying to do something with Manage Allocations, add two fields to it and when you start recreating your own page you run into other dependencies.  Thank you for any insight anyone may have.

**Comment by coriobriensfdo** [comment]
@stephaniepck we don't provide support here in GitHub, so my recommendation would be to post in the [Power of Us Hub](https://powerofus.force.com/s/group/0F980000000CjReCAK/nonprofit-success-pack) and ask for advice.

**Comment by eblazer** [comment]
@stephaniepck We did the same thing. We've rebuilt the page by cloning code from this repo, then added-in support for a Field Set. It did take a bit of effort to get all the dependencies worked out, but in the end is working.

@coriobriensfdo Would love to talk about getting this included in the base package so that others don't need to go through this headache. Let me know if you're interested.

**Comment by coriobriensfdo** [comment]
[Work Item to resolve](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008IoKVIA0/view) (community contribution)
